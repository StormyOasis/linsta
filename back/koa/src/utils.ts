import jwt from "jsonwebtoken";
import { Context } from "koa";

import {
  config,
  DBConnector,
  EDGE_POST_TO_COMMENT,
  EDGE_USER_FOLLOWS,
  EDGE_USER_LIKED_POST,
  getPfpByUserId,
  logger,
  type Like,
  type Post,
  handleSuccess as sharedHandleSuccess,
  handleValidationError as sharedHandleValidationError
} from "@linsta/shared";

export const isUserAuthorized = (ctx: Context, userId: string): boolean => {
    return ctx.state.user?.id === userId;
}

export const handleSuccess = (ctx: Context, result: unknown) => {
    const data = sharedHandleSuccess(result);

    ctx.status = data.statusCode;
    ctx.body = data.body;
    ctx.set(data.headers);
}

export const handleValidationError = (ctx: Context, error: string, statusCode: number = 400) => {
    const data = sharedHandleValidationError(error, statusCode);

    ctx.status = data.statusCode;
    ctx.body = data.body;
    ctx.set(data.headers);
}

export type JWTData = {
    id: string;
}

export const verifyJWT = async (ctx: Context, next: () => unknown) => {
    const authHeader: string | undefined = ctx.request.headers['authorization'];
    let token: string | undefined = undefined;
    if (authHeader) {
        const bearer = authHeader.split(' ');
        token = bearer[1];
    }

    if (!token) {
        ctx.res.statusCode = 403;
        ctx.body = { status: "Invalid token" };
        return;
    }

    let result = null;
    await jwt.verify(
        token as string,
        config.auth.jwt.secret,
        (err, decoded) => {
            const data = decoded as JWTData;
            if (err || decoded == null) {
                ctx.res.statusCode = 403;
                ctx.body = { status: "Invalid token" };
                return;
            }
            result = data.id;
        })

    if (result == null) {
        ctx.res.statusCode = 403;
        ctx.body = { status: "Error verifying id" };
        return;
    }

    ctx.res.statusCode = 200;
    ctx.body = { status: "OK" };

    await next();
}

export const addPfpsToPosts = async (posts: Record<string, Post>) => {
    await Promise.all(Object.values(posts).map(async post => {
        post.user.pfp = await getPfpByUserId(false, post.user.userId);
    }));
}

export const addCommentCountsToPosts = async (posts: Record<string, Post>, postIds: string[]) => {
    const __ = DBConnector.__();
    const commentResults = await (await DBConnector.getGraph()).V(postIds)
        .project("postId", "commentCount")
        .by(__.id())
        .by(__.outE(EDGE_POST_TO_COMMENT).count())
        .toList();

    for (const result of commentResults) {
        const data = DBConnector.unwrapResult(result);
        const parsed = DBConnector.parseGraphResult<{ postId: string; commentCount: number }>(
            data,
            ["postId", "commentCount"]
        );

        if (parsed.postId in posts) {
            posts[parsed.postId].global.commentCount = parsed.commentCount;
        }
    }
};

export const addLikesToPosts = async (posts: Record<string, Post>, postIds: string[]) => {
    const __ = DBConnector.__();
    const likeResults = await (await DBConnector.getGraph()).V(postIds)
        .filter(__.inE(EDGE_USER_LIKED_POST).count().is(DBConnector.P().gt(0)))
        .project("postId", "users")
        .by(__.id())
        .by(__.inE(EDGE_USER_LIKED_POST)
            .outV()
            .project('profileId', 'userName', 'pfp', 'firstName', 'lastName', 'userId')
            .by("profileId")
            .by("userName")
            .by("pfp")
            .by("firstName")
            .by("lastName")
            .by(__.id())
            .fold())
        .toList();

    for (const result of likeResults) {
        const data = DBConnector.unwrapResult(result);
        const parsed = DBConnector.parseGraphResult<{ postId: string; users: unknown[] }>(data, ["postId", "users"]);

        const { postId, users } = parsed;

        if (!posts[postId]) {
            continue;
        }

        const post = posts[postId];
        post.global.likes = [];

        for (const user of users ?? []) {
            const userData = DBConnector.unwrapResult(user);
            const userParsed = DBConnector.parseGraphResult<Like>(userData, [
                "userName",
                "userId",
                "profileId",
                "firstName",
                "lastName",
                "pfp"
            ]);

            post.global.likes.push({
                userName: userParsed.userName,
                userId: userParsed.userId,
                profileId: userParsed.profileId,
                firstName: userParsed.firstName,
                lastName: userParsed.lastName,
                pfp: userParsed.pfp
            });
        }
    }
};

export const getFollowingUserIds = async (userId: string): Promise<string[]> => {
    try {
        const results = await (await DBConnector.getGraph())
            .V(userId)
            .out(EDGE_USER_FOLLOWS)
            .toList();

        return results.map(result => {
            const vertex = DBConnector.unwrapResult(result);
            if (vertex instanceof Map) {
                return String(vertex.get("id"));
            } else if (typeof vertex === "object" && vertex !== null && "id" in vertex) {
                return String((vertex as { id: unknown }).id);
            }
            throw new Error("Unexpected vertex format");
        });

    } catch (err) {
        logger.error("Failed to get following user IDs", { userId, error: err });
        return [];
    }
}