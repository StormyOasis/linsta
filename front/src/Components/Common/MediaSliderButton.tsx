import React from 'react';
import styled from 'styled-components';

import { Div, Flex } from './CombinedStyling';
import { LeftArrowSVG, RightArrowSVG } from './Icon';

const MediaSliderWrapper = styled(Div)`
    z-index: 20;
    position: absolute;
    bottom: 0;
`;

const MediaSliderLeftNormalWrapper = styled(MediaSliderWrapper)`
    bottom: 43%;
    left: -1%;
`;

const MediaSliderRightNormalWrapper = styled(MediaSliderWrapper)`
    bottom: 43%;    
`;

const MediaSliderBottomLeftWrapper = styled(MediaSliderWrapper)`
    right: 44px;
`;

const MediaSliderBottomRightWrapper = styled(MediaSliderWrapper)`
    right: 5px;
`;

const MediaSliderButtonContainer = styled(Flex)`
    width: 24px;
    height: 24px;
    color: ${props => props.theme['colors'].borderDefaultColor};
    background-color: ${props => props.theme['colors'].cropperAspectBkgnd};
    border-radius: 50%;
    padding: 5px; 
    cursor: pointer;
    
    &:hover {
        color: ${props => props.theme['colors'].borderDefaultColor};
        background-color: ${props => props.theme['colors'].cropperAspectBkgndNoTrans};
        border: 1px solid ${props => props.theme['colors'].borderDarkColor};
    };    
`;

type MediaSliderButtonProps = {
    direction: ("left" | "right" | "bottomLeft" | "bottomRight");
    onClick: () => void;
};

const MediaSliderButton: React.FC<MediaSliderButtonProps> = (props: MediaSliderButtonProps) => {
    const renderLeft = () => {
        return (
            <MediaSliderLeftNormalWrapper>
                <MediaSliderButtonContainer onClick={props.onClick}>
                    <LeftArrowSVG width="24px" height="24px" fill="currentColor" stroke="none" />
                </MediaSliderButtonContainer>
            </MediaSliderLeftNormalWrapper>
        );
    }
    
    const renderRight = () => {
        return (
            <MediaSliderRightNormalWrapper>
                <MediaSliderButtonContainer onClick={props.onClick}>
                    <RightArrowSVG width="24px" height="24px" fill="currentColor" stroke="none" />
                </MediaSliderButtonContainer>
            </MediaSliderRightNormalWrapper>
        );
    }

    const renderBottomLeft = () => {
        return (
            <MediaSliderBottomLeftWrapper>
                <MediaSliderButtonContainer onClick={props.onClick}>
                    <LeftArrowSVG width="24px" height="24px" fill="currentColor" stroke="none" />
                </MediaSliderButtonContainer>
            </MediaSliderBottomLeftWrapper>
        );        
    }

    const renderBottomRight = () => {
        return (
            <MediaSliderBottomRightWrapper>
                <MediaSliderButtonContainer onClick={props.onClick}>
                    <RightArrowSVG width="24px" height="24px" fill="currentColor" stroke="none" />
                </MediaSliderButtonContainer>
            </MediaSliderBottomRightWrapper>
        );        
    }

    return (
        <>
            {props.direction === "left" && renderLeft()}
            {props.direction === "right" && renderRight()}
            {props.direction === "bottomLeft" && renderBottomLeft()}
            {props.direction === "bottomRight" && renderBottomRight()}
        </>
    );
}

export default MediaSliderButton;