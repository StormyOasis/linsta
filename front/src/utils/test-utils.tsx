import React, { ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { RootState } from "../Components/Redux/redux";
import { authInitialState } from '../Components/Redux/slices/auth.slice';
import { modalInitialState } from '../Components/Redux/slices/modals.slice';
import { profileInitialState } from '../Components/Redux/slices/profile.slice';
import { miscInitialState } from '../Components/Redux/slices/misc.slice';
import { configureStore } from '@reduxjs/toolkit';

import Theme from '../Components/Themes/Theme';

export const mockInitialState = (overrides?: Partial<RootState>): RootState => ({
    auth: { ...authInitialState, ...(overrides?.auth ?? {}) },
    modal: { ...modalInitialState, ...(overrides?.modal ?? {}) },
    profile: { ...profileInitialState, ...(overrides?.profile ?? {}) },
    misc: { ...miscInitialState, ...(overrides?.misc ?? {}) },
});

type ExtendedRenderOptions = {
    store?: ReturnType<typeof configureStore>;
    preloadedState?: any;
    route?: string; // ‚Üê allow overriding the initial route
} & Omit<RenderOptions, "queries">;

const dummyAuthReducer = (state = { user: null, isLoading: false, error: null }, _action: any) => state;
const dummyModalReducer = (state = { isOpen: false }, _action: any) => state;
const dummyProfileReducer = (state = { name: '', email: '' }, _action: any) => state;
const dummyMiscReducer = (state = { someData: null }, _action: any) => state;

const renderWithStore = (
    ui: ReactElement,
    {
        preloadedState,
        store = configureStore({
            reducer: {
                auth: dummyAuthReducer,
                modal: dummyModalReducer,
                profile: dummyProfileReducer,
                misc: dummyMiscReducer,
            },
            preloadedState,
        }),
        route = "/",
        ...renderOptions
    }: ExtendedRenderOptions = {}
) => {
    const Wrapper: React.FC<{ children: React.ReactNode }> = ({ children }) => {
        return (
            <Provider store={store}>
                <MemoryRouter initialEntries={[route]}>
                    <Theme>{children}</Theme>
                </MemoryRouter>
            </Provider>
        );
    };

    return {
        ...render(ui, { wrapper: Wrapper, ...renderOptions }),
        store,
    };
};

export * from "@testing-library/react";
export { renderWithStore };
export type { ExtendedRenderOptions };