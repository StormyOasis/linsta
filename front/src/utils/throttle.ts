import { useRef } from 'react';

const useThrottle = (callback: Function, delay: number) => {
    const lastCall = useRef(0);

    return (...args: any[]) => {
        const now = new Date().getTime();
        if (now - lastCall.current >= delay) {
            lastCall.current = now;
            callback(...args);
        }
    };
};

export default useThrottle;