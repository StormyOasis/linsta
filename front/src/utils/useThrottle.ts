import { useRef, useCallback } from 'react';

/**
 * React hook for throttling a callback function.
 * The callback will only be invoked at most once every `delay` milliseconds.
 *
 * @param callback - The function to throttle.
 * @param delay - The minimum delay (ms) between calls.
 * @returns A throttled version of the callback.
 */
const useThrottle = <T extends (...args: any[]) => void>(callback: T, delay: number) => {
    const lastCall = useRef<number>(0);

    return useCallback(
        (...args: Parameters<T>) => {
            const now = performance.now();
            if (now - lastCall.current >= delay) {
                lastCall.current = now;
                callback(...args);
            }
        },
        [callback, delay]
    );
};

export default useThrottle;