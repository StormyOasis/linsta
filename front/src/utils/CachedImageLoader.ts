import { clear, del, get, set } from 'idb-keyval';
import { blobToBase64 } from "./utils";

export const loadImageCached = async (id: string, url: string): Promise<string> => {
    // First try to get the image from the cache
    const value = await get(id);

    if(value === undefined) {
        // Image not in the cache so have to fetch it
        try {
            const result = await fetch(url);
            if(result.status === 200) {
                const blob = await result.blob();
                const b64:string = await blobToBase64(blob) as string;

                // add the base64 data into the cache and return it
                await set(id, b64);

                return b64;
            }
        } catch(err) {
            console.log(err);
            return url; //default to what was passed in
        }

    } else {
        // Image is in the cache already as a base64 string, use that as is
        return value;
    }

    return url; //default to what was passed in
}

export const setImage = async (id: string, data: string) => {     
    await set(id, data);
}

export const deleteFromCache = async (id: string) => {
    // First try to get the image from the cache
    const value = await get(id);

    if(value === undefined) {
        // key not present in cache so do nothing
        return;
    }

    // delete from the cache
    await del(id);
}

export const clearCache = async () => {
    await clear();
}