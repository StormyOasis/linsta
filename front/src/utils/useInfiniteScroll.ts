import React, { useEffect, useRef } from 'react';
import useThrottle from './useThrottle';

/**
 * Type for the async load more function.
 */
type AsyncLoadMoreFunction = () => Promise<void>;

/**
 * Custom hook for infinite scrolling.
 * Attaches a throttled scroll event listener to a container and triggers `loadMore`
 * when the user scrolls near the bottom.
 *
 * @param loadMore - Async function to load more items.
 * @param isLoading - Whether a load is currently in progress.
 * @param containerRef - Ref to the scrollable container element.
 */
const useInfiniteScroll = (loadMore: AsyncLoadMoreFunction, 
    isLoading: boolean, 
    childRef: React.MutableRefObject<HTMLDivElement | null>) => {
        
    const hasScrolled = useRef<boolean>(false);  // Track if the user has scrolled

    // Throttled scroll handler
    const throttledHandleScroll = useThrottle(async () => {
        if (typeof window !== 'undefined' && childRef.current) {
            const element = childRef.current as HTMLElement;
            const currentScroll = window.innerHeight + element.scrollTop;

            if (currentScroll + 256 >= element.scrollHeight && !isLoading && !hasScrolled.current) {
                hasScrolled.current = true;  // Mark as scrolled at least once
                await loadMore();
            }
        }
    }, 200);

    useEffect(() => {
        const childElement = childRef.current;
        if (childElement) {
            childElement.addEventListener('scroll', throttledHandleScroll);
        }

        // Clean up event listener on unmount
        return () => {
            if (childElement) {
                childElement.removeEventListener('scroll', throttledHandleScroll);
            }
        };
    }, [throttledHandleScroll, childRef]);

    // Reset trigger flag when loading finishes
    useEffect(() => {
        if (!isLoading) {
            hasScrolled.current = false;
        }
    }, [isLoading]);
}

export default useInfiniteScroll;