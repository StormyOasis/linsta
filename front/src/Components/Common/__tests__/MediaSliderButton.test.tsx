import React from "react";
import { screen, fireEvent } from "@testing-library/react";
import MediaSliderButton from "../MediaSliderButton";
import '@testing-library/jest-dom';
import { renderWithStore } from '../../../utils/test-utils';

// Mock icons
jest.mock("../Icon", () => ({
    LeftArrowSVG: (props: any) => <svg data-testid="left-arrow" {...props} />,
    RightArrowSVG: (props: any) => <svg data-testid="right-arrow" {...props} />,
}));

describe("MediaSliderButton", () => {
    it("renders left direction and calls onClick", () => {
        const handleClick = jest.fn();
        const r = renderWithStore(<MediaSliderButton direction="left" onClick={handleClick} />);
        const leftArrow = screen.getByTestId("left-arrow");
        expect(leftArrow).toBeInTheDocument();
        fireEvent.click(leftArrow.parentElement!);
        expect(handleClick).toHaveBeenCalled();
        expect(r).toMatchSnapshot();
    });

    it("renders right direction and calls onClick", () => {
        const handleClick = jest.fn();
        const r = renderWithStore(<MediaSliderButton direction="right" onClick={handleClick} />);
        const rightArrow = screen.getByTestId("right-arrow");
        expect(rightArrow).toBeInTheDocument();
        fireEvent.click(rightArrow.parentElement!);
        expect(handleClick).toHaveBeenCalled();
        expect(r).toMatchSnapshot();
    });

    it("renders bottomLeft direction and calls onClick", () => {
        const handleClick = jest.fn();
        const r = renderWithStore(<MediaSliderButton direction="bottomLeft" onClick={handleClick} />);
        const leftArrow = screen.getByTestId("left-arrow");
        expect(leftArrow).toBeInTheDocument();
        fireEvent.click(leftArrow.parentElement!);
        expect(handleClick).toHaveBeenCalled();
        expect(r).toMatchSnapshot();
    });

    it("renders bottomRight direction and calls onClick", () => {
        const handleClick = jest.fn();
        const r = renderWithStore(<MediaSliderButton direction="bottomRight" onClick={handleClick} />);
        const rightArrow = screen.getByTestId("right-arrow");
        expect(rightArrow).toBeInTheDocument();
        fireEvent.click(rightArrow.parentElement!);
        expect(handleClick).toHaveBeenCalled();
        expect(r).toMatchSnapshot();
    });
});