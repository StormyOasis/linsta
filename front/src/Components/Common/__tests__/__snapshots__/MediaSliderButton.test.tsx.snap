// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediaSliderButton renders bottomLeft direction and calls onClick 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="CombinedStyling__Div-sc-c3jdxk-0 MediaSliderButton__MediaSliderWrapper-sc-9298gd-0 MediaSliderButton__MediaSliderBottomLeftWrapper-sc-9298gd-3 escqfF jYODlU ieIDDl"
      >
        <div
          class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 MediaSliderButton__MediaSliderButtonContainer-sc-9298gd-5 escqfF ckFvxY juDGix"
        >
          <svg
            data-testid="left-arrow"
            fill="currentColor"
            height="24px"
            stroke="none"
            width="24px"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="CombinedStyling__Div-sc-c3jdxk-0 MediaSliderButton__MediaSliderWrapper-sc-9298gd-0 MediaSliderButton__MediaSliderBottomLeftWrapper-sc-9298gd-3 escqfF jYODlU ieIDDl"
    >
      <div
        class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 MediaSliderButton__MediaSliderButtonContainer-sc-9298gd-5 escqfF ckFvxY juDGix"
      >
        <svg
          data-testid="left-arrow"
          fill="currentColor"
          height="24px"
          stroke="none"
          width="24px"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`MediaSliderButton renders bottomRight direction and calls onClick 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="CombinedStyling__Div-sc-c3jdxk-0 MediaSliderButton__MediaSliderWrapper-sc-9298gd-0 MediaSliderButton__MediaSliderBottomRightWrapper-sc-9298gd-4 escqfF jYODlU EejTc"
      >
        <div
          class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 MediaSliderButton__MediaSliderButtonContainer-sc-9298gd-5 escqfF ckFvxY juDGix"
        >
          <svg
            data-testid="right-arrow"
            fill="currentColor"
            height="24px"
            stroke="none"
            width="24px"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="CombinedStyling__Div-sc-c3jdxk-0 MediaSliderButton__MediaSliderWrapper-sc-9298gd-0 MediaSliderButton__MediaSliderBottomRightWrapper-sc-9298gd-4 escqfF jYODlU EejTc"
    >
      <div
        class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 MediaSliderButton__MediaSliderButtonContainer-sc-9298gd-5 escqfF ckFvxY juDGix"
      >
        <svg
          data-testid="right-arrow"
          fill="currentColor"
          height="24px"
          stroke="none"
          width="24px"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`MediaSliderButton renders left direction and calls onClick 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="CombinedStyling__Div-sc-c3jdxk-0 MediaSliderButton__MediaSliderWrapper-sc-9298gd-0 MediaSliderButton__MediaSliderLeftNormalWrapper-sc-9298gd-1 escqfF jYODlU jIczEF"
      >
        <div
          class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 MediaSliderButton__MediaSliderButtonContainer-sc-9298gd-5 escqfF ckFvxY juDGix"
        >
          <svg
            data-testid="left-arrow"
            fill="currentColor"
            height="24px"
            stroke="none"
            width="24px"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="CombinedStyling__Div-sc-c3jdxk-0 MediaSliderButton__MediaSliderWrapper-sc-9298gd-0 MediaSliderButton__MediaSliderLeftNormalWrapper-sc-9298gd-1 escqfF jYODlU jIczEF"
    >
      <div
        class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 MediaSliderButton__MediaSliderButtonContainer-sc-9298gd-5 escqfF ckFvxY juDGix"
      >
        <svg
          data-testid="left-arrow"
          fill="currentColor"
          height="24px"
          stroke="none"
          width="24px"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`MediaSliderButton renders right direction and calls onClick 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="CombinedStyling__Div-sc-c3jdxk-0 MediaSliderButton__MediaSliderWrapper-sc-9298gd-0 MediaSliderButton__MediaSliderRightNormalWrapper-sc-9298gd-2 escqfF jYODlU dDOaem"
      >
        <div
          class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 MediaSliderButton__MediaSliderButtonContainer-sc-9298gd-5 escqfF ckFvxY juDGix"
        >
          <svg
            data-testid="right-arrow"
            fill="currentColor"
            height="24px"
            stroke="none"
            width="24px"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="CombinedStyling__Div-sc-c3jdxk-0 MediaSliderButton__MediaSliderWrapper-sc-9298gd-0 MediaSliderButton__MediaSliderRightNormalWrapper-sc-9298gd-2 escqfF jYODlU dDOaem"
    >
      <div
        class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 MediaSliderButton__MediaSliderButtonContainer-sc-9298gd-5 escqfF ckFvxY juDGix"
      >
        <svg
          data-testid="right-arrow"
          fill="currentColor"
          height="24px"
          stroke="none"
          width="24px"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
